src/Models/UserModel.php

```diff
@@ -208,4 +210,41 @@ private function checkQueryReturn(bool $return): void
             throw new RuntimeException($message, $error['code']);
         }
     }
+
+    /**
+     * Override the BaseModel's `save()` method to allow
+     * updating of user email, password, or password_hash
+     * fields if they've been modified.
+     *
+     * @param mixed $data
+     */
+    public function save($data): bool
```

Comment:
Not `mixed` but `User`?

```suggestion
    public function save(User $data): bool
```

Reply:
If we specify `User` then we won't be able to save arrays. I would recommend the docblock be specified though: `User|array`

Reply:
Then it also complains about not being compatible with the method that we're overriding.